heron/spi/tests/java/com/twitter/heron/spi/utils/ShellUtilsTest.java

```diff
@@ -88,15 +88,25 @@ public void testLargeOutput() throws IOException, InterruptedException {
 
   @Test
   public void testRunAsyncProcess() throws IOException {
-    String testString = "testString";
-    StringBuilder stdout = new StringBuilder();
-    StringBuilder stderr = new StringBuilder();
-    // Sleep 1 second and echo some text.
-    Process p = ShellUtils.runASyncProcess(
-        String.format("sleep 1 && echo %s", testString));
-    // Test process is running and input stream is empty
-    wait(10, TimeUnit.MILLISECONDS);
-    Assert.assertEquals(0, p.getInputStream().available());
```

Comment:
This old test actually didn't test the behavior. `sleep 1 && echo ...` will be split to `sleep`, `1`, `&&`, `echo`, `...`, and the code will launch the process `sleep` with the other strings as its argument. Then `sleep` will fail at once. Since `redirectErrorStream` didn't work before, the error info will go into stderr. So `p.getInputStream().available()` will always be 0.